### ISSUE TITLE
Release MetaCLIPv2 on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [facebookresearch/MetaCLIP#78]
Hello @howardhsu 🤗 

I'm Niels and work as part of the open-source team at Hugging Face. I made MetaCLIP available on Hugging Face last year, see https://huggingface.co/facebook/metaclip-h14-fullcc2.5b for instance. I also made a [tweet](https://x.com/NielsRogge/status/1717259646602236136) that went pretty viral :D

I just indexed the paper page of MetaCLIPv2: https://huggingface.co/papers/2410.17251. The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance) you can also claim the paper as yours which will show up on your public profile at HF.

Would you like to host the v2 models you've pre-trained on https://huggingface.co/models, rather than Github? Hosting on Hugging Face will give you more visibility/enable better discoverability. We can add tags in the model cards so that people find the models easier, link it to the paper page, etc.

If you're down, leaving a guide [here](https://huggingface.co/docs/hub/models-uploading). After uploaded, we can also link the models to the paper page (read [here](https://huggingface.co/docs/hub/en/model-cards#linking-a-paper)) so people can discover your model.

Let me know if you need any help!

Cheers,
Niels

**Comment by NielsRogge** [comment]
Hi @howardhsu gentle reminder on this :)

Happy to assist with uploading models. MetaCLIP v1 is pretty popular on 🤗 !

**Comment by howardhsu** [comment]
thx for reminding and appreciated for promoting Meta CLIP 1.0. We do have
1.2 (renamed from 2.0) released on README, as part of NeurIPS bandle from
FAIR. We also have bigG model as 1.1 released end of last year i guess not
in Huggingface yet. Let me know what is needed for integrated in
Huggingface.

Homepage: https://howardhsu.github.io/
Linkedin: https://www.linkedin.com/in/hu-xu-9852403b/
Google Scholar: https://scholar.google.com/citations?user=SaH2yWMAAAAJ
Twitter: https://twitter.com/Hu_Hsu
Email: ***@***.***

**Comment by NielsRogge** [comment]
Are the MetaCLIP 2.0 checkpoints compatible with the original CLIP architecture?

Cause that's what we did for v1, we just converted the weights to the Transformers format: https://huggingface.co/models?search=metaclip.

We could do the same for v2

**Comment by howardhsu** [comment]
the arch is available on huggingface:

for Meta CLIP 1.2(please avoid using MetaCLIP2.0) ViT-H/14 we only change
quickgelu to nn.gelu for scalability and observed no perf. diffs.

the ViT bigG model (Meta CLIP 1.1) released last year still uses quickgelu.



best
Hu

Homepage: https://howardhsu.github.io/
Linkedin: https://www.linkedin.com/in/hu-xu-9852403b/
Google Scholar: https://scholar.google.com/citations?user=SaH2yWMAAAAJ
Twitter: https://twitter.com/Hu_Hsu
Email: ***@***.***

**Comment by howardhsu** [comment]
how's integration going on @NielsRogge now?

**Comment by NielsRogge** [comment]
Hi @howardhsu, to convert the MetaCLIPv2 weights, you can use the conversion script: https://gist.github.com/Natyren/47e564357cac95bf923b7c65781492c2. Based on this thread: https://github.com/huggingface/transformers/issues/26584.

**Comment by swdanielli** [comment]
hi @NielsRogge  thank you so much for helping us pushing the ckpts to HF! We would definitely love to transfer it to Facebook namespace. I wonder if you can help us with that?

**Comment by NielsRogge** [comment]
Sure, do you already have a HF account? If yes, then you could join the "Facebook" organization on Hugging Face. I'll share the conversion script with you.

It would also be nice if you and your co-authors could claim the paper: https://huggingface.co/papers/2507.22062.

In the meantime, the second model has been pushed: https://huggingface.co/nielsr/metaclip-2-huge-worldwide-378 nielsr/metaclip-2-huge-worldwide-378 · Hugging Face

**Comment by NielsRogge** [comment]
Thanks, I'll let you know when the Transformers PR is merged.

**Comment by NielsRogge** [comment]
Hi @howardhsu, all checkpoints have been pushed to my personal user account: https://huggingface.co/models?search=nielsr%20worldwide. Would you be able to make them available at the `facebook` organization on the hub?

**Comment by NielsRogge** [comment]
Ok thanks, btw I can just use `inputs = tokenizer(text, padding=True)` - there's no need to pad the the max length of 77 right?

I'll take care of the namespace change.

**Comment by howardhsu** [comment]
If the encode_text can handle truncating extra position_emeddings, then yes (no need to pad to 77 tokens). Thanks for taking care and let us know if any issue.
